build/docker/package.sh

```diff
@@ -38,7 +39,7 @@ done
 gzip -f datadog-dotnet-apm.tar
 
 if [ -z "${MUSL-}" ]; then
-  mv datadog-dotnet-apm.tar.gz datadog-dotnet-apm-$VERSION.tar.gz
+  mv datadog-dotnet-apm.tar.gz datadog-dotnet-apm-$VERSION.$ARCH.tar.gz
```

Comment:
Small note: This will change the name of the x64 file but I think the change to include the arch name is what we want. We'll need to possibly update our public docs and maybe our in-app docs as well. I've added the `needs-docs-update` label to the PR

Reply:
For users who script their downloads, this is a breaking change. If we want to rename it, we can consider a migration path where we have both names for some time, update the docs, add notes to releases for some number of releases, etc. What do you think?

Reply:
I agree this could be a breaking change, I'll revert this filename change on `x64` versions.

Reply:
done in 3684467